### ISSUE TITLE
Bug: lang() placeholders don't work with quotes

### ISSUE THREAD

**Issue by Ryubal** [codeigniter4/CodeIgniter4#4913]
**Describe the bug**
When attempting to use lang() with a string that contains placeholders surrounded by quotes (single or double), they don't get replaced.

**CodeIgniter 4 version**
4.1.3

**Affected module(s)**
..

**Expected behavior, and steps to reproduce if appropriate**
Inside of a translations file (e.g. Language/en/Errors.php) add a new string with a placeholder surrounded by quotes:
```php
return [
	'username_exists' => "Username '{username}' already exists"
]
```

Then, attempt to print it in a controller response:
```php
public function create() {
	...
	if($userExists)
		return $this->failResourceExists(lang('Errors.username_exists', ['username' => 'myUsername']));
}
```

Expected response: **Username 'myUsername' already exists**
Actual response: **Username {username} already exists**

Also tried using double quotes in a string made of single quotes, or escaped doubled/single quotes, and nothing worked.

**Context**
 - OS: Windows 10
 - Web server: Apache
 - PHP version: 7.4.3

**Comment by michalsn** [comment]
Under the hood, we use `MessageFormatter::formatMessage` so I can only guess that this is restricted for some reason. I haven't dig into it though. This should work for you:

```php
return [
	'username_exists' => 'Username "{username}" already exists'
]
```

**Comment by sfadschm** [comment]
I am not using named placeholders, but using 

    'delete'  => '"{0}" wurde erfolgreich gelöscht.'

works fine for me in my projects.

**Comment by Ryubal** [comment]
Thank you both for your insights.

I did some digging into `MessageFormatter:formatMessage` and I found out that single quotes are used as escaping characters. So if someone wants to have them in the final output, they must be escaped using another single quote. Example:

```php
'username_exists' => "Username ''{username}'' already exists"
```

I added this as a note to the [PHP docs](https://www.php.net/manual/en/messageformatter.formatmessage.php)
